### DISCUSSION TITLE
How to do multiple crop and move operations in one invocation?

### DISCUSSION THREAD

**Discussion by pvonmoradi** [ImageMagick/ImageMagick#4887] [Help] [answered]
I have an input image. I want to:
1- crop to part of the image to `new_bg`
2- crop another part of the original image, resize and move and overlay it onto `new_bg` -> `new_bg2`
3- crop another part of the original image, resize and move and overlay it onto `new_bg2` -> `new_bg3`

Here is what I did:
```
# first I cut the pieces I want
    convert in.jpg -crop 380x370+0+31 +repage green_bg.jpg
    convert in.jpg -crop 12x58+625+17 +repage -resize 105x yellow.jpg
    convert in.jpg -crop 38x73+233+47 +repage -resize 90x red.jpg
# then I overlay pieces step by step
    composite -compose over -gravity south-west yellow.jpg green_bg.jpg yellow_on_green_bg.jpg
    composite -compose over -gravity south-east red.jpg yellow_on_green_bg.jpg yellow_and_red_on_green_bg.jpg
```

This does what I want. Can I somehow do all of it in one invocation? (Ignore the dimensions and sizes of the crop, they are not correct).
Here is a sample input:
![image](https://user-images.githubusercontent.com/1058151/155332291-faef1f2b-7d7b-4b62-bc60-31d36dcb71b0.png)

Here is what I want:
![image](https://user-images.githubusercontent.com/1058151/155332586-a7c50939-033b-4c95-abcd-15f83e8103bd.png)

`Linux`
`Version: ImageMagick 6.9.7-4 Q16 x86_64 20170114 http://www.imagemagick.org`

**Comment by fmw42** [comment, marked as answer]
You can use parentheses processing to do each part. See https://legacy.imagemagick.org/Usage/basics/#parenthesis and https://legacy.imagemagick.org/Usage/files/#mpr

```
convert in.jpg -write mpr:img +delete \
\( mpr:img -crop 380x370+0+31 +repage \) \
\( mpr:img -crop 12x58+625+17 +repage -resize 105x \) \
-gravity southwest -composite \
\( mpr:img -crop 38x73+233+47 +repage -resize 90x \) \
-gravity southeast -composite \
yellow_and_red_on_green_bg.jpg

```
